mmcv/cnn/rfsearch/search.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import logging
+import os
+from typing import Dict, Optional
+
+import torch  # noqa
+import torch.nn as nn
+
+import mmcv
+from mmcv.runner import HOOKS, Hook
+from .operator import Conv2dRFSearchOp, ConvRFSearchOp  # noqa
+from .utils import write_to_json
+
+logging.basicConfig(
+    format='[%(asctime)s-%(filename)s-%(levelname)s:%(message)s]',
+    level=logging.NOTSET,
+    datefmt='%Y-%m-%d%I:%M:%S %p',
+)
+logger = logging.getLogger('Searcher')
+logger.setLevel(logging.ERROR)
+
+
+@HOOKS.register_module()
+class RFSearchHook(Hook):
+    """Rcecptive field search via dilation rates.
+        Paper: RF-Next: Efficient Receptive Field
+            Search for Convolutional Neural Networks, TPAMI2022 (CVPR2021)
+            https://arxiv.org/abs/2206.06637
+
+    Args:
+        mode (str, optional):
+                mode can be set to the following types:
+                 search/fixed_single_branch/fixed_multi_branch.
```

Comment:
```suggestion
        mode (str, optional): It can be set to the following types:
            'search', 'fixed_single_branch', or 'fixed_multi_branch'.
```